project/src/test/java/com/google/sps/test/PlacesFetcherTest.java

```diff
@@ -0,0 +1,177 @@
+
+  /** A valid PlacesSearchResult with placeId "PLACEID_1". */
+  private static final PlacesSearchResult SEARCH_RESULT_1 = createTestPlacesSearchResult(PLACEID_1);
+
+  /** A valid PlacesSearchResult with placeId "PLACEID_2". */
+  private static final PlacesSearchResult SEARCH_RESULT_2 = createTestPlacesSearchResult(PLACEID_2);
+
+  /** An array of valid PlacesSearchResults. */
+  private static final PlacesSearchResult[] SEARCH_RESULT_ARR = {SEARCH_RESULT_1, SEARCH_RESULT_2};
+
+  /** Valid PlaceDetails for PLACE_1. */
+  private static final PlaceDetails PLACE_DETAILS_1 =
+    createTestPlaceDetails("name1", URL, PHONE, RATING, PRICELEVEL, LOCATION);
+
+  /** Valid PlaceDetails for PLACE_2. */
+  private static final PlaceDetails PLACE_DETAILS_2 =
+    createTestPlaceDetails("name2", URL, PHONE, RATING, PRICELEVEL, LOCATION);
+
+  /**
+   * creates URL from String for PlaceDetails used in tests.
+   *
+   * @param s The string representing the URL
+   * @return A URL
+   */
+  private static URL createTestURL(String s) {
+    try {
+      return new URL(s);
+    } catch (MalformedURLException e) {
+      System.out.println("url not generated");
+      return null;
+    }
+  }
+
+  /**
+   * creates PlaceDetails to be used in tests.
+   *
+   * @param name The name of the Place
+   * @param url The url of the Place's website
+   * @param phone The phonenumber of the Place
+   * @param rating The Google rating of the Place
+   * @param priceLevel The priceLevel of the Place as identified by Google Places
+   * @param location The location of the Place
+   * @return The PlacesDetails according to the given params
+   */
+  private static PlaceDetails createTestPlaceDetails(String name, URL url, String phone,
+      float rating, PriceLevel priceLevel, LatLng location) {
+    PlaceDetails placeDetails = new PlaceDetails();
+    placeDetails.name = name;
+    placeDetails.website = url;
+    placeDetails.formattedPhoneNumber = phone;
+    placeDetails.rating = rating;
+    placeDetails.priceLevel = priceLevel;
+    placeDetails.geometry = new Geometry();
+    placeDetails.geometry.location = location;
+    return placeDetails;
+  }
+
+  /** creates PlaceSearchResult to be used in tests.
+   *
+   * @param placeId The place Id of the created PlaceSearchResult
+   * @return PlacesSearchResult with placeId as it's Id
+   */
+  private static PlacesSearchResult createTestPlacesSearchResult(String placeId) {
+    PlacesSearchResult searchResult = new PlacesSearchResult();
+    searchResult.placeId = placeId;
+    return searchResult;
+  }
+
+  /** A PlacesFetcher instance to be spied on. */
+  private static PlacesFetcher placesFetcher = new PlacesFetcher();
+
+  @Test
+  public void fetch_zeroSearchResults_returnsEmptyList() throws Exception {
+    PlacesFetcher spiedFetcher = spy(placesFetcher);
+    doReturn(new PlacesSearchResult[0]).when(spiedFetcher).getPlacesSearchResults();
```

Comment:
Interesting solution! 
The problem that I see with the approach you took here, is that you won't be able to test the `getPlacesSearchResult` function. 
Instead, I think that you should have a single method that only calls `request.await()` and returns the value. Then you can mock out this method (and this method only), but still test your other method (the one in which you choose inputs). 
Same goes for the place details (you can even combine them). 
LMK if you want to talk about this f2f.

Reply:
I understand and will change it. 
Just one dillema - do you think I should make an interface with a test implementation where those request.await() methods will be public so I can mock them, and production implementaion where they are private? Or it doesn't matter for them to be public?

Reply:
Sound good, but this means the test has to be in the same package as PlaceFetcher and it's not how our packages are orginized. So you think I should put them in the same package or just use the annotation but with public visibility?

Reply:
I changed the functions like you suggested but I ran into a problem with testing the requests sent to the methods that call request.await() so IMO it's still not good enough. I pushed the best thing I got to and we can discuss this f2f and I explain the problem.